### ISSUE TITLE
Generating UUID in freeze_time context

### ISSUE THREAD

**Issue by mhq** [spulec/freezegun#154]
There seems to be an issue with generating UUIDs when the freeze time is set to a future time. For some reason, all UUIDs generated outside the freeze context will be associated with the frozen timestamp.  However, the context seems to be released properly when the freeze time is set to the past. 
Any ideas?

Here are some tests to illustrate (The second unit test will also fail if both tests are run in sequence.):

``` python
import uuid
from datetime import datetime, timedelta
from cassandra.util import datetime_from_uuid1
from freezegun import freeze_time

# Fails. uuid1() will generate the same timestamp uuid after the freeze context.
# assert datetime.datetime(2016, 9, 15, 13, 27, 48, 527016) > 
#   datetime.datetime(2016, 9, 15, 13, 27, 48, 527016)
def test_uuid_with_future_freeze_time():
    with freeze_time(datetime.utcnow() + timedelta(hours=2)):
        uuid1 = uuid.uuid1()

    uuid2 = uuid.uuid1()
    assert datetime_from_uuid1(uuid1) > datetime_from_uuid1(uuid2)


# Passes
def test_uuid_with_past_freeze_time():
    with freeze_time(datetime.utcnow() - timedelta(hours=2)):
        uuid1 = uuid.uuid1()

    uuid2 = uuid.uuid1()
    assert datetime_from_uuid1(uuid1) < datetime_from_uuid1(uuid2)
```

**Comment by spulec** [comment]
Thanks for opening. This is an interesting issue.

Reading through the [`uuid1` code](https://hg.python.org/cpython/file/default/Lib/uuid.py), it seems to not work when moving time backwards. Notice how `_last_timestamp` is used.

I suppose we could file a bug with python. I'm not sure if they will consider this a valid use case.

You can work around this by setting `uuid._last_timestamp = None`.

**Comment by romuald** [comment]
I don't seem to have this issue, as uuid1 does not seems to be patched by freezegun at all (always return the current time)

(your test passes)

- python 2.7
- freezegun 0.3.7 and 0.3.8

**Comment by spulec** [comment]
This should now be fixed with #172
